tests/e2e-tests/specs/front-end/front-end-checkout.test.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * @format
+ */
+
+/**
+ * Internal dependencies
+ */
+import { createSimpleProduct } from '../../utils/components';
+import { CustomerFlow, StoreOwnerFlow } from '../../utils/flows';
+import { setCheckbox, settingsPageSaveChanges, uiUnblocked, verifyCheckboxIsSet } from '../../utils';
+
+const config = require( 'config' );
+
+let orderId;
+
+describe( 'Checkout page', () => {
+	beforeAll( async () => {
+		await StoreOwnerFlow.login();
+		await createSimpleProduct();
+
+		// Go to general settings page
+		await StoreOwnerFlow.openSettings( 'general' );
+
+		// Set base location with state CA.
+		await expect( page ).toSelect( 'select[name="woocommerce_default_country"]', 'United States (US) — California' );
+		// Sell to all countries
+		await expect( page ).toSelect( '#woocommerce_allowed_countries', 'Sell to all countries' );
+		// Set currency to USD
+		await expect( page ).toSelect( '#woocommerce_currency', 'United States (US) dollar ($)' );
+		// Tax calculation should have been enabled by another test - no-op
+		// Save
+		await settingsPageSaveChanges();
+
+		// Verify that settings have been saved
+		await Promise.all( [
+			expect( page ).toMatchElement( '#message', { text: 'Your settings have been saved.' } ),
+			expect( page ).toMatchElement( 'select[name="woocommerce_default_country"]', { text: 'United States (US) — California' } ),
+			expect( page ).toMatchElement( '#woocommerce_allowed_countries', { text: 'Sell to all countries' } ),
+			expect( page ).toMatchElement( '#woocommerce_currency', { text: 'United States (US) dollar ($)' } ),
+		] );
+
+		// Enable BACS payment method
+		await StoreOwnerFlow.openSettings( 'checkout', 'bacs' );
+		await setCheckbox( '#woocommerce_bacs_enabled' );
+		await settingsPageSaveChanges();
+
+		// Verify that settings have been saved
+		await verifyCheckboxIsSet( '#woocommerce_bacs_enabled' );
+
+		// Enable COD payment method
+		await StoreOwnerFlow.openSettings( 'checkout', 'cod' );
+		await setCheckbox( '#woocommerce_cod_enabled' );
+		await settingsPageSaveChanges();
+
+		// Verify that settings have been saved
+		await verifyCheckboxIsSet( '#woocommerce_cod_enabled' );
+
+		// Enable PayPal payment method
+		await StoreOwnerFlow.openSettings( 'checkout', 'paypal' );
+		await setCheckbox( '#woocommerce_paypal_enabled' );
+		await settingsPageSaveChanges();
+
+		// Verify that settings have been saved
+		await verifyCheckboxIsSet( '#woocommerce_paypal_enabled' );
+
+		await StoreOwnerFlow.logout();
+	} );
```

Comment:
I see how settings can be configured differently but not sure what do you mean by "add some kind of first step in that direction". I imagine we'd add whatever we need when we'd be writing new tests so that it could be tested right away?

Reply:
You're right, let's leave this for when we actually add further tests.